rfcs/0140-simple-package-paths.md

```diff
@@ -0,0 +1,270 @@
+    - It also slows down or even locks up editors due to the file size
+  - `nix edit -f . package-attr` works, though that's not yet stable (it relies on the `nix-command` feature being enabled) and doesn't work with packages that don't set `meta.position` correctly).
+- `all-packages.nix` frequently causes merge conflicts. It's a point of contention for all new packages
+
+# Detailed design
+[design]: #detailed-design
+
+This RFC consists of two parts, each of which is implemented with a PR to Nixpkgs.
+
+## PR 1: The unit directory standard
+
+This part establishes the new _unit directory standard_ in Nixpkgs.
+This standard is internal to Nixpkgs and not exposed as public interface.
+This standard must be documented in the Nixpkgs manual.
+
+### File structure
+
+Create the initially-empty directory `pkgs/unit`, called _unit base directory_, in Nixpkgs.
+
+Check the following using CI:
+- The unit base directory must only contain subdirectories of the form `pkgs/unit/${shard}/${name}`, called _unit directories_.
+- `name` is a string only consisting of the ASCII characters a-z, A-Z, 0-9, `-` or `_`.
+- `shard` is the lowercased first two letters of `name`, expressed in Nix: `shard = toLower (substring 0 2 name)`.
+- Each unit directory must contain a `package.nix` file and may contain arbitrary other files.
+
+### Semantics
+
+Introduce code to automatically define `pkgs.${name}` for each unit directory as a value equivalent to
+```nix
+pkgs.callPackage pkgs/unit/${shard}/${name}/package.nix { }`
+```
+
+Check the following using CI for each unit directory:
+- The only definition for `pkgs.${name}` is the automatically generated one from the unit directory
+- `pkgs.${name}` must evaluate to a [derivation](https://nixos.org/manual/nix/stable/glossary.html#gloss-derivation).
+- The `package.nix` file must not transitively refer to files outside its unit directory.
+
+## PR 2: Migration
+
+Automatically migrate to the unit directory standard for all definitions `pkgs.${name}` that can be migrated by
+- Only moving files
+- Not changing the evaluation result
+
+This will cause merge conflicts with all existing PRs that modify such moved files, however they can trivially be rebased using `git rebase && git push -f`.
+However, to have the least amount of conflicts, this migration should be performed soon after a release when ZHF is over and the PR rate slows down.
+This also gives a lot of time to fix any potential problems before the next release.
+
+Manual updates may also be done to ensure further non-evaluation validity, such as
+- [CODEOWNERS](https://github.com/NixOS/nixpkgs/blob/master/.github/CODEOWNERS)
+- Update scripts like [this](https://github.com/NixOS/nixpkgs/blob/cb2d5a2fa9f2fa6dd2a619fc3be3e2de21a6a2f4/pkgs/applications/version-management/cz-cli/generate-dependencies.sh)
+
+Due to the strict limitations of standard, this PR will not start enforcing it for new packages.
```

Comment:
Since [custom arguments are allowed again](https://github.com/NixOS/rfcs/pull/140#discussion_r1191584512), this isn't necessary anymore. I re-added the requirement that new packages must adhere to the standard.